pubtools/_pulp/tasks/push/items/base.py

```diff
@@ -140,6 +145,146 @@ def upload_context(cls, pulp_client):
         """
         return UploadContext(client=pulp_client)
 
+    @classmethod
```

Comment:
Note, all this code has just been moved from command.py but otherwise has no substantial changes.